presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -366,6 +372,112 @@ else if (firstDependencies.stream().allMatch(right::canResolve) && secondDepende
         return new RelationPlan(root, analysis.getScope(node), outputSymbols);
     }
 
+    private RelationPlan planJoinUsing(Join node, RelationPlan left, RelationPlan right)
```

Comment:
please take a look at my suggestion at the top of `Analysis.java`. If we go that way we may just need to add a code for that would "project" join output symbols instead of creating whole new code path for join using.